### ISSUE TITLE
ServerRejectedTlsHandshake when running --reneg and --resum

### ISSUE THREAD

**Issue by svengo** [nabla-c0d3/sslyze#488]
**Describe the bug**
Two errors while scanning an nginx server.

**To Reproduce**
1. Install SSLyze using pip
2. Run the following command `sslyze --regular arnor.svengo.net`

**Expected behavior**
The scan should complete without an error :-)

**Python environment (please complete the following information):**
 - OS: `Ubuntu 20.04.2 LTS` under kvm
 - `Linux docker-test01 5.4.0-62-generic #70-Ubuntu SMP Tue Jan 12 12:45:47 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`
 - Python version: 3.8.5

**Additional context**
The server is running  `nginx/1.14.2` with built with `OpenSSL 1.1.1d` and `ssl_protocols TLSv1.3`.

```
cloud-user@docker-test01:~$ sslyze arnor.svengo.net
...
 * Error when running --reneg:
       You can open an issue at https://github.com/nabla-c0d3/sslyze/issues with the following information:

       * Server: arnor.svengo.net:443 - 5.135.143.191
       * Scan command: session_renegotiation

       Traceback (most recent call last):
         File "/home/cloud-user/.local/lib/python3.8/site-packages/sslyze/scanner.py", line 264, in get_results
    result = implementation_cls.result_for_completed_scan_jobs(
         File "/home/cloud-user/.local/lib/python3.8/site-packages/sslyze/plugins/session_renegotiation_plugin.py", line 100, in result_for_completed_scan_jobs
    result_enum, value = job.result()
         File "/usr/lib/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
         File "/usr/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
         File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
         File "/home/cloud-user/.local/lib/python3.8/site-packages/sslyze/plugins/session_renegotiation_plugin.py", line 122, in _test_secure_renegotiation
    ssl_connection.connect()
         File "/home/cloud-user/.local/lib/python3.8/site-packages/sslyze/connection_helpers/tls_connection.py", line 342, in connect
    raise ServerRejectedTlsHandshake(
       sslyze.errors.ServerRejectedTlsHandshake

 * Error when running --resum:
       You can open an issue at https://github.com/nabla-c0d3/sslyze/issues with the following information:

       * Server: arnor.svengo.net:443 - 5.135.143.191
       * Scan command: session_resumption

       Traceback (most recent call last):
         File "/home/cloud-user/.local/lib/python3.8/site-packages/sslyze/scanner.py", line 264, in get_results
    result = implementation_cls.result_for_completed_scan_jobs(
         File "/home/cloud-user/.local/lib/python3.8/site-packages/sslyze/plugins/session_resumption/implementation.py", line 230, in result_for_completed_scan_jobs
    result_enum, value = job.result()
         File "/usr/lib/python3.8/concurrent/futures/_base.py", line 432, in result
    return self.__get_result()
         File "/usr/lib/python3.8/concurrent/futures/_base.py", line 388, in __get_result
    raise self._exception
         File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
         File "/home/cloud-user/.local/lib/python3.8/site-packages/sslyze/plugins/session_resumption/_resumption_with_id.py", line 56, in resume_with_session_id
    session1 = resume_tls_session(server_info, tls_version_to_use)
         File "/home/cloud-user/.local/lib/python3.8/site-packages/sslyze/plugins/session_resumption/_resumption_with_id.py", line 35, in resume_tls_session
    ssl_connection.connect()
```

**Comment by UncleRaymondo** [comment]
I have the same issue and it occurs when TLS 1.2 is disabled and only TLS1.3 is enabled. These commands are specific to TLS 1.2.

**Comment by nabla-c0d3** [comment]
Fix released as part of [v4.0.3](https://github.com/nabla-c0d3/sslyze/releases/tag/4.0.3).

**Comment by svengo** [comment]
Thank you for the quick fix, @nabla-c0d3!
